layers/+lang/c-c++/README.org

```diff
@@ -52,22 +66,134 @@ by setting the variable =c-c++-default-mode-for-headers= to =c++-mode=.
+- Semantic highlighting
+- See more on [[https://github.com/cquery-project/cquery/wiki/Emacs]]
+- Cross-platform - functional on Windows, Linux and OSX.
+
+**** External dependencies
+Install one (or both) of the following:
+
+***** cquery server
+Install the =cquery= server. [[https://github.com/cquery-project/cquery/wiki/Getting-started][Instructions]].
+
+***** ccls server
+Install the =ccls= server. [[https://github.com/MaskRay/ccls/wiki/Getting-started][Instructions]].
+
+**** Configuration
+***** Basic
+Select either =cquery= or =ccls= as the =c-c++= layer backend by adding the following to your dotfile:
+
+#+BEGIN_SRC emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
+                '((c-c++ :variables c-c++-backend `lsp-cquery))) ;or 'lsp-ccls
+#+END_SRC
+
+N.B. The [[../../+tools/lsp/README.org][LSP layer]] will be loaded automatically if either backend is selected.
+
+***** Setting path to backend executable
+The basic configuration above should work if the cquery/ccls executable folder is present in your path. If not, you can set the path explicitly.
+
+#+BEGIN_SRC emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
+                '((c-c++ :variables
+                         c-c++-backend `lsp-cquery
+                         c-c++-lsp-executable "/path/to/bin/cquery/or/ccls")))
+#+END_SRC
+
+If you need to expand =~= in the path, you can use =file-truename= like
+
+#+BEGIN_SRC emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
+                '((c-c++ :variables
+                         c-c++-backend `lsp-cquery
+                         c-c++-lsp-executable (file-truename "~/bin/cquery/or/ccls"))))
+#+END_SRC
+
+***** Semantic highlighting
+Semantic highlighting is disabled by default. To enable, set the =c-c++-lsp-sem-highlight-method= variable to either ='font-lock= or ='overlay=.
+To enable the rainbow semantic highlighting colour theme, set =c-c++-lsp-sem-highlight-rainbow= to =t=.
+
+***** Additional configuration options
+Both lsp backends are configured to store their index cache in a subdirectory of =.emacs.d/cache=. This can be overridden by
+specifying an explicit =c-c++-lsp-cache-dir=. Setting this value to a relative path will cause the index cache to be placed in a
+subdirectory of your project root.
+
+There are other initialization options such as the number of indexer threads, cache serialization format.
+They have good default values. See [[file:./config.el][config.el]] and the backends' respective homepages for more info.
+- [[https://github.com/cquery-project/cquery/wiki/Emacs][Emacs section of =cquery= wiki]]
+- [[https://github.com/MaskRay/ccls/wiki/Emacs][Emacs section of =ccls= wiki]]
+
+***** Example dotspacemacs-configuration-layers entry
+#+BEGIN_SRC emacs-lisp
+  (setq-default dotspacemacs-configuration-layers
+                '((c-c++ :variables
+                         c-c++-adopt-subprojects t
+                         c-c++-backend 'ccls
```

Comment:
Thanks svenihoney - updated the readme. @sdwolfz , you happy enough to use the 'Apply suggestion' feature as a general rule? Looks very convenient for this sort of change (oversights in the README). I'd imagine it probably results in a new commit though rather than an amendment (which appears to be the recommended approach in the beginner's guide to PRs)? Regardless, went with the traditional `git commit --amend`, `git push --force` in this instance, pending enlightenment :)

Reply:
It's a nice feature but I would like to have as few commits as possible at the end. So if they are small fixups then I'd preffer they be squashed anyway.